cooker/lib/pages.js

```diff
@@ -241,58 +392,27 @@ async function pages(file, localeFile) {
 
   logger.start("Generating pages");
 
-  // go throught all of them
+  // go through all of them
   let promises = [];
 
   for (let file of pugFiles) {
     // get the file information and locale files
     let templateInfo = path.parse(file);
 
-    // The data file for MD. This just includes any object data within an MD file, if that's what we're compiling
-    const mdData = {};
+    // Container variable for markdown data
+    let mdData;
 
-    // Otherwise compile as a pug file
     // if file starts with underscore, we ignore it, expected as standard 👍
     if (templateInfo.name.charAt(0) == "_") continue;
 
     // If we're a markdown file, compile with grey-matter
+    // This is a great pattern that we can just extend as we need to to render different kinds of files.
     if (templateInfo.ext.toLowerCase() === ".md") {
-      const mdfile = matter.read(file);
-      const name = templateInfo.name;
-      const dir = templateInfo.dir;
-      const templateFile = path.join(templateInfo.dir, mdfile.data.template);
-
-      // If we have a template file defined in the grey-matter file
-      if (mdfile.data.template) {
-        // If the template doesn't exist, warn and continuer
-        if (!fse.pathExistsSync(templateFile)) {
-          logger.warning([
-            `Error compiling ${path.relative(
-              config.CWD,
-              file
-            )}. The markdown file calls for the template ${path.relative(
-              config.CWD,
-              templateFile
-            )} which does not exist`,
-          ]);
-          continue;
-        } else {
-          // Fill out the markdown data content
-          // Convert the content using markdown.
-          mdData.content = Markdown.render(mdfile.content);
-          // Transcribe the returned data components to the markdown data object
-          for (let t in mdfile.data) {
-            mdData[t] = mdfile.data[t];
-          }
-          // Parse the template file
-          templateInfo = path.parse(templateFile);
-          // ... But set the name of the output to the name of the md file
-          templateInfo.name = name;
-          // ... and set the dir to the md file dir
-          templateInfo.dir = dir;
-          // Finally update the file to parse to the provided template file.
-          file = templateFile;
-        }
+      try {
+        ({ templateInfo, file, mdData } = parseMarkdown(file, templateInfo));
```

Comment:
Don't know if the `parseMarkdown` function should mutate `templateInfo` like this. This might be a personal preference but I think that those last steps...
```js
// Parse the template file
      templateInfo = path.parse(templateFile);
      // ... But set the name of the output to the name of the md file
      templateInfo.name = name;
      // ... and set the dir to the md file
      templateInfo.dir = dir;
```
...should be done in here, after you return the `mdData` as they are pertinent to the template and not the markdown data. What do you think?

Reply:
The issue here is that we want the template to be passed in the same way that a regular pug page would be, right? Because the template that renders one of these md files is just pug. The reason this is done this way is so that this information and data is passed to the template, which then goes through and does what these pages normally does - consuming model data, a11y etc.

Reply:
@marlonmarcello I can talk you through my reasoning here on a call at some point if you want.

Reply:
I understood what i was doing. This is minor and more of a code organization preference, in my opinion changing the template path isn't something that `parseMarkdown` should do. It should __just__ parse it and return everything so the compile function can decide what to do with it.

Reply:
Ah I see what you mean. That's a good call and something I'll look into.

Reply:
I've updated this, take a look.

Reply:
Looking great @liamegan 
Thanks for that.